oxidation/libparsec/crates/types/src/regex.rs

```diff
@@ -12,43 +31,135 @@ fn escape_globing_pattern(string: &str) -> String {
 }
 
 impl Regex {
-    pub fn from_pattern(pattern: &str) -> Result<Self, fnmatch_regex::error::Error> {
-        let escaped_str = escape_globing_pattern(pattern);
-        Ok(Regex(fnmatch_regex::glob_to_regex(&escaped_str)?))
+    /// Returns a regex which is built from a file that contains shell like patterns
+    pub fn from_file(file_path: &Path) -> RegexResult {
+        let file_content =
+            fs::read_to_string(file_path).map_err(|err| RegexError::PatternFileIOError { err })?;
+        let mut regexes = Vec::new();
+        for pattern_line in file_content.lines() {
+            // Skip comment and empty line
+            if pattern_line.trim().starts_with('#') || pattern_line.trim() == "\n" {
+                continue;
+            }
+
+            let escaped_glob = escape_globing_pattern(pattern_line.trim());
+            regexes.push(
+                fnmatch_regex::glob_to_regex(&escaped_glob)
+                    .map_err(|err| RegexError::ShellPatternError { err })?,
+            );
+        }
```

Comment:
For the challenge, if you want:

You can rewrite this block of code without using a `for-loop` but by using the `Iterator` trait

Reply:
Done: https://github.com/Scille/parsec-cloud/pull/3737/commits/c2e5e1720cd3bedb1ce2551632e715a543bde89f